rust/worker/src/segment/types.rs

```diff
@@ -440,26 +441,20 @@ impl<'me> LogMaterializer<'me> {
             "Total length of logs in materializer: {}",
             self.logs.total_len()
         );
-        let next_offset_id;
-        match self.curr_offset_id.as_ref() {
+        let next_offset_id = match self.curr_offset_id.as_ref() {
             Some(curr_offset_id) => {
-                next_offset_id = curr_offset_id.clone();
-                next_offset_id.fetch_add(1, std::sync::atomic::Ordering::SeqCst);
+                curr_offset_id.fetch_add(1, std::sync::atomic::Ordering::SeqCst);
+                curr_offset_id.clone()
```

Comment:
I usually leave a comment when dealing with cur/next as to whether the variable is set to the last-returned value, or the next-to-generate value.